### PR TITLE
Temperature sensor, data status, and more

### PR DESCRIPTION
**Pull request by jessebraham**
Bunch of miscellaneous changes here:

- Added missing license files and README
- Temperature sensor support
- Data status checks
- Various low-hanging fruit
- Improve clarity (function renaming, fewer magic numbers, etc.)

### CODE REVIEW

File: src/register.rs
Comment (sajattack) [inline src/register.rs:187]: Can you explain what these mean a bit better?

**Review by sajattack — CHANGES_REQUESTED** [review]

File: src/register.rs
Comment (jessebraham) [reply src/register.rs:187]: This was again just taken from the `lis2dh12` library, so I'm not too attached to it. I agree it's not super clear without context.

`STATUS_REG` has the following bit fields:

* `ZYXOR` - X, Y and Z-axis data overrun
* `ZOR` - Z-axis data overrun
* `YOR` - Y-axis data overrun
* `XOR` - X-axis data overrun
* `ZYXDA` - X, Y and Z-axis new data available
* `ZDA` - Z-axis new data available
* `YDA` Y-axis new data available
* `XDA` X-axis new data available

So this struct basically just splits the fields up into groups for convenience:

`zyxor` -> `ZYXOR`  
`xyzor` -> (`XOR`, `YOR`, `ZOR`)  
`zyxda` -> `ZYXDA`  
`xyzda` -> (`XDA`, `YDA`, `ZDA`)  

I have no problem changing it if there's a better way to do this, just seemed reasonable enough to me that I didn't spend too much time thinking about it.

File: src/lib.rs
Comment (jacobrosenthal) [inline src/lib.rs]: is_in_boot is more idiomatic for booleans

Comment (jessebraham) [reply src/lib.rs]: Good eye! I'll get that changed, thanks.

File: src/register.rs
Comment (jacobrosenthal) [reply src/register.rs:187]: I can see it being fine, I dont think most people are going to touch it. However I closed my PR here and PRed this branch with my convenience function that is the more common representation just checking ZYXDA. The idea is if you're polling the sensor for data as fast as possible, you need to actually wait until it updates ZYXDA before you try to read fresh data.

Comment (sajattack) [reply src/register.rs:187]: Yeah, doc comments with the explanation you just provided would be sufficient.

Comment (jacobrosenthal) [reply src/register.rs:187]: So if you're only reading x, you only need to wait til x is ready. 

But our common use case is raw_accell grabbing all three so code in a tight loop SHOULD be doing
```
            while !lis3dh.is_data_ready().unwrap() {}
            let dat = lis3dh.accel_raw().unwrap();
```


My pr waiting for @jessebraham  to approve is here
https://github.com/jessebraham/lis3dh-rs/pull/1

Comment (jacobrosenthal) [reply src/register.rs:187]: Oh and overflow means that two reading wrote over eachother before you read were able to read. Most people dont care, but I could see adding an `is_overflowed()` or something in the future which would be just ZYXOR

Comment (jessebraham) [reply src/register.rs:187]: @jacobrosenthal thanks for the contribution.
I've added the doc comment.

Comment (sajattack) [reply src/register.rs:187]: Any reason not to bake the call to `is_data_ready` into `accel_raw`?

Comment (jacobrosenthal) [reply src/register.rs:187]: Its not a terrible Idea, I just dont think Ive seen anyone else do it so im not sure its idiomatic for accel_raw...

Well I searched github, theres actually one guy doing it
https://github.com/JosephBGerber/lis3dsh-rs/blob/5ef60fa5d95199df18fcfb20d6ea1a2e810c36b7/src/lib.rs

It adds an extra round trip which might annoy some zero costers. And people might be confused and not understand to check for a WouldBlock thats not actually an error...

Might be appropriate to bring up in an issue in the accelerometer trait crate.

Comment (jacobrosenthal) [reply src/register.rs:187]: https://github.com/NeoBirth/accelerometer.rs/issues/61

Comment (jessebraham) [reply src/register.rs:187]: I haven't seen other crates doing that. My gut is leaning towards maintaining the separation of concerns here, but I don't feel too strongly either way about it.

File: src/lib.rs
Comment (jacobrosenthal) [inline src/lib.rs:328]: norm calculation needs work I think. I not sure why its shifting by 4 at all?
We need to fix up sensitivity for all the different modes
shifting right divides by 2^n so better to do it that way than a floating?
>>1 /2
>>2 /4
>>3 /8
>>4 /16
>>5 /32
>>6 /64

then accel_norm wants in G units so mg * 1000=g
From datasheet
// 2g high res  1mg/digit	>>0 * 1000
// 2g normal    4mg/digit	>>2 * 1000
// 2g low       16mg/digit	>>4 * 1000

// 4g high res  2mg/digit	>>1 * 1000
// 4g normal    8mg/digit	>>3 * 1000
// 4g low       32mg/digit	>>5 * 1000

// 8g high res  4mg/digit	>>2 * 1000
// 8g normal    16mg/digit	>>4 * 1000
// 8g low       64mg/digit	>>6 * 1000

12 48, 192 dont have a cheat
// 16g high res  12mg/digit	 * 1000/12
// 16g normal    48mg/digit	 * 1000/48
// 16g low       192mg/digit	 * 1000/192

File: src/lib.rs
Comment (jacobrosenthal) [inline src/lib.rs:114]: 3.2.1 table 11 says theres a turn on time to switching into modes
Might want to busy wait or 
12-bit mode to 8-bit mode1/ODR
12-bit mode to 10-bit mode1/ODR
10-bit mode to 8-bit mode1/ODR
10-bit mode to 12-bit mode7/ODR
8-bit mode to 10-bit mode1/ODR
8-bit mode to 12-bit mode7/ODR
(ODR is the data rate)

File: src/lib.rs
Comment (jacobrosenthal) [inline src/lib.rs:137]: The data rate and mode are intermingled for certain data rates.

The easiest fix is to just remove 1.6, 1.344 and 5.376 rates because they cant actually be set independently from a mode. Or we could check and error in the data rate (and mode!) setters.

Or we could have mode and rate be set with the same function, maybe even the same enum like HR1hz, Low1hz, normal1hz, low5.376hz.  Or maybe an low(1hz), low(5.376hz)

8.8 Table 31
ODR
0000 Power-down mode
0001 HR / Normal / Low-power mode (1 Hz)
0010 HR / Normal / Low-power mode (10 Hz)
0011 HR / Normal / Low-power mode (25 Hz)
0100 HR / Normal / Low-power mode (50 Hz)
0101 HR / Normal / Low-power mode (100 Hz)
0110 HR / Normal / Low-power mode (200 Hz)
0111 HR / Normal / Low-power mode (400 Hz)
1000 Low power mode (1.60 kHz)
1001HR / normal (1.344 kHz);Low-power mode (5.376 kHz)

File: src/lib.rs
Comment (jacobrosenthal) [inline src/lib.rs:69]: Why are we mucking with High res mode outside of set_mode?

Comment (jessebraham) [reply src/lib.rs:69]: I'm not entirely sure, there was a bunch of initialization in the constructor and I had just left it in place. I think a lot of the settings are default values anyways, so I'm not sure if it's appropriate to have them in there or not. Might be better to supply a `reset` function or something of that nature.

Comment (jacobrosenthal) [reply src/lib.rs:69]: Yeah these definitely arent your changes. But they're definitely not default values.

Comment (jessebraham) [reply src/lib.rs:69]: Just FYI, I've given you access to my fork if you're interested. I am looking into the issues, but feel free to do whatever.

**Comment by jacobrosenthal** [comment]
(not that Im a committer here :)  ) but I say this can be merged as is if no one has problems other than what I see, as most of what I see was preexisting. Then Ill make a pr with the issues im finding and we can talk about how or if to do breaking changes fixing it up?

**Pull request merged**
